src/transformers/models/clip/modeling_clip.py

```diff
@@ -858,13 +874,29 @@ def forward(
         )
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
-        encoder_states = () if output_hidden_states else None
+        if isinstance(output_hidden_states, list):
+            output_hidden_states = [
+                min(self.config.num_hidden_layers, self.config.num_hidden_layers + index + 1) if index < 0 else index
+                for index in output_hidden_states
+            ]
+
+        if output_hidden_states:
+            if isinstance(output_hidden_states, list):
+                encoder_states = list(output_hidden_states)
+            else:
+                encoder_states = ()
+        else:
+            encoder_states = None
```

Comment:
nit, we can streamline a bit
```suggestion
        if isinstance(output_hidden_states, list):
            encoder_states = list(output_hidden_states)
        elif output_hidden_states:
            encoder_states = ()
        else:
            encoder_states = None
```